### PR TITLE
234 azure blob storage service

### PR DESCRIPTION
**Pull request by serhiiyasenev**
dev
## JIRA

* [Main JIRA ticket 234](https://github.com/project-studying-dotnet/Streetcode-Server-June/issues/234)

1. Cloud
image

3. Local storage (emulator)
image

### CODE REVIEW

File: Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            using (var stream = new MemoryStream(bytes))
+            {
+                blobClient.Upload(stream, new BlobHttpHeaders { ContentType = $"application/{extension}" });
+            }
+        }
+
+        public void DeleteFileInStorage(string name)
+        {
+            var blobClient = _containerClient.GetBlobClient(name);
+            blobClient.DeleteIfExists();
```
Comment (metrondir) [inline Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs:72]: make method async

File: Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        }
+
+        public void SaveFileInStorageBase64(string base64, string name, string extension)
+        {
+            var blobClient = _containerClient.GetBlobClient($"{name}.{extension}");
+            var bytes = Convert.FromBase64String(base64);
+
+            using (var stream = new MemoryStream(bytes))
+            {
+                blobClient.Upload(stream, new BlobHttpHeaders { ContentType = $"application/{extension}" });
```
Comment (metrondir) [inline Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs:65]: make method async

File: Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        }
+
+        public string SaveFileInStorage(string base64, string name, string mimeType)
+        {
+            var blobClient = _containerClient.GetBlobClient(name);
+
+            var bytes = Convert.FromBase64String(base64);
+            using (var stream = new MemoryStream(bytes))
+            {
+                blobClient.Upload(stream, new BlobHttpHeaders { ContentType = mimeType });
```
Comment (metrondir) [inline Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs:52]: make method async

File: Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+            _containerClient = _blobServiceClient.GetBlobContainerClient(environment.ContainerName);
+            _containerClient.CreateIfNotExists();
+            _repositoryWrapper = repositoryWrapper;
+        }
+
+        public MemoryStream FindFileInStorageAsMemoryStream(string name)
+        {
+            var blobClient = _containerClient.GetBlobClient(name);
+            var downloadInfo = blobClient.DownloadContent();
```
Comment (metrondir) [inline Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs:31]: make method async

File: Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+            var memoryStream = new MemoryStream(downloadInfo.Value.Content.ToArray());
+            return memoryStream;
+        }
+
+        public string FindFileInStorageAsBase64(string name)
+        {
+            var blobClient = _containerClient.GetBlobClient(name);
+            var downloadInfo = blobClient.DownloadContent();
```
Comment (metrondir) [inline Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs:39]: make method async

File: Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        public void DeleteFileInStorage(string name)
+        {
+            var blobClient = _containerClient.GetBlobClient(name);
+            blobClient.DeleteIfExists();
+        }
+
+        public string UpdateFileInStorage(string previousBlobName, string base64Format, string newBlobName, string extension)
+        {
+            DeleteFileInStorage(previousBlobName);
+            var newBlobUri = SaveFileInStorage(base64Format, newBlobName, extension);
```
Comment (metrondir) [inline Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs:78]: make method async

**Review by metrondir — APPROVED** [review]
lgtm!

File: Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+namespace Streetcode.BLL.Services.BlobStorageService
+{
+    public class AzureBlobService : IBlobService
+    {
+        private readonly BlobServiceClient _blobServiceClient;
+        private readonly BlobContainerClient _containerClient;
+        private readonly IRepositoryWrapper _repositoryWrapper;
+
+        public AzureBlobService(IOptions<BlobEnvironmentVariables> options, IRepositoryWrapper? repositoryWrapper = null)
```
Comment (metrondir) [inline Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs:15]: why repositoryWrapper = null ?

Comment (serhiiyasenev) [reply Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs:15]: it doesn't matter, but updated

File: Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs (hunk 4 above)
Comment (serhiiyasenev) [reply Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs:31]: it is synced in `IBlobService` and we should implement the interface

File: Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs (hunk 5 above)
Comment (serhiiyasenev) [reply Streetcode/Streetcode.BLL/Services/BlobStorageService/AzureBlobService.cs:39]: it is synced in `IBlobService` and we should implement the interface

**Review by IrynaNakoneshniuk — APPROVED** [review]

**Pull request merged**